### PR TITLE
Add budgeted campaigns in order to manage budget

### PR DESCRIPTION
**Pull request by coderberry**
(ch164)

### CODE REVIEW

File: lib/code_fund/schemas/budgeted_campaign.ex [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  This schema is for a postgres materialized view
+  """
+
+  use CodeFundWeb, :schema
+
+  @primary_key false
+  @foreign_key_type :binary_id
+  schema "budgeted_campaigns" do
+    belongs_to(:campaign, CodeFund.Schema.Campaign)
+    belongs_to(:user, CodeFund.Schema.User)
```
Comment (StareIntoTheBeard) [inline lib/code_fund/schemas/budgeted_campaign.ex]: Do we need all of these relationships (aside from User)? Looks like we just query this on campaign only.

File: lib/code_fund/schemas/user.ex [hunk 2]
```diff
@@ -10,6 +10,7 @@ defmodule CodeFund.Schema.User do
     has_many(:campaigns, CodeFund.Schema.Campaign)
     has_many(:properties, CodeFund.Schema.Property)
     has_many(:creatives, CodeFund.Schema.Creative)
+    has_many(:budgeted_campaigns, CodeFund.Schema.BudgetedCampaign)
```
Comment (StareIntoTheBeard) [inline lib/code_fund/schemas/user.ex]: Are we planning on querying the user's budgeted campaigns?

File: priv/repo/migrations/20180702160920_create_campaign_budgets_materialized_view.exs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        )
+        SELECT
+          *,
+          (total_spend - revenue_amount) AS balance,
+          ((revenue_amount / total_impressions) * 1000) AS actual_ecpm
+        FROM data;
+    """)
+
+    create(unique_index("budgeted_campaigns", [:campaign_id]))
+    create(index("budgeted_campaigns", [:user_id]))
```
Comment (StareIntoTheBeard) [inline priv/repo/migrations/20180702160920_create_campaign_budgets_materialized_view.exs]: I don't think we need most of these indices, other than campaigns,

File: lib/code_fund/schemas/budgeted_campaign.ex (hunk 1 above)
Comment (coderberry) [reply lib/code_fund/schemas/budgeted_campaign.ex]: Probably not. I will remove them.

File: lib/code_fund/schemas/user.ex (hunk 2 above)
Comment (coderberry) [reply lib/code_fund/schemas/user.ex]: Yes

File: priv/repo/migrations/20180702160920_create_campaign_budgets_materialized_view.exs (hunk 3 above)
Comment (coderberry) [reply priv/repo/migrations/20180702160920_create_campaign_budgets_materialized_view.exs]: It will be in metabase

**Pull request merged**